### DISCUSSION TITLE
No module named 'arabert.preprocess_arabert'

### DISCUSSION THREAD

**Discussion by jubaer145** [aub-mind/arabert#65] [Q&A] [answered]
Hi,
When I am running the jupyter notebook you provided, `AraBERT-PyTorch-Demo.ipynb`, I am getting `No module named 'arabert.preprocess_arabert'`  error.

I am not sure how to use the 'arabert.preprocess_arabert'`. 

Please help.

**Comment by WissamAntoun** [comment, marked as answer]
I replaced it with this code 

```
from arabert.preprocess import ArabertPreprocessor

model_name = "bert-base-arabertv2"
arabert_prep = ArabertPreprocessor(model_name=model_name, keep_emojis=False)

text = "ولن نبالغ إذا قلنا إن هاتف أو كمبيوتر المكتب في زمننا هذا ضروري"
arabert_prep.preprocess(text)
>>>"و+ لن نبالغ إذا قل +نا إن هاتف أو كمبيوتر ال+ مكتب في زمن +نا هذا ضروري"
```

Also, I recommend you don't use `AraBERT-PyTorch-Demo.ipynb` since this is a very old code. Please use one of the new examples in the colabs

**Reply by jubaer145** [reply]
got it! Thanks a lot!!

**Reply by Astudnew** [reply]
Please , in this part , did you use Farasa?
"text = "ولن نبالغ إذا قلنا إن هاتف أو كمبيوتر المكتب في زمننا هذا ضروري"
arabert_prep.preprocess(text)
>>>"و+ لن نبالغ إذا قل +نا إن هاتف أو كمبيوتر ال+ مكتب في زمن +نا هذا ضروري""

**Reply by WissamAntoun** [reply]
yes, whenever you see a "+" in an example then the model is using Farasa

**Reply by manelAffi** [reply]
@WissamAntoun 
Hello,
 I'm beginner with arabert.
 even when I use the above code the error still persist.

 ```
from arabert.preprocess import ArabertPreprocessor

ModuleNotFoundError:` No module named 'arabert
```
Please advise.
Also, I want to know if  there is any tutorial how to include arabert as word embedding layer with customer  model using Keras for named entity recognition task?
I will be grateful for any help.

**Reply by WissamAntoun** [reply]
Hey, I think the file that you are using to call `from arabert.preprocess import ArabertPreprocessor` is in the arabert folder. You can either create the file outside of the arabert folder or change it to `from preprocess import ArabertPreprocessor`.

As for the tutorial using Keras and BERT as an embedding layer only, I have yet to come by one. By I think you might find some online for sure
